### PR TITLE
mcs: refactor tso handler

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Need to merge #6963 first.

```commit-message
```

### Check List

Tests

- Unit test
- Manual test

```
tiup-playground nightly --pd.mode ms
```

try small one:
```
curl -X POST 127.0.0.1:2379/pd/api/v1/admin/reset-ts --data '{"tso":"329750308454400000"}'
[PD:tso:ErrResetUserTimestamp]reset user timestamp failed, the specified ts is smaller than now%
```

try large one:
```
curl -X POST 127.0.0.1:2379/pd/api/v1/admin/reset-ts --data '{"tso":"919750308454400000", "force-use-larger":true}'
Reset ts successfully.%
```
pd log:
```
[2023/08/16 18:30:50.801 +08:00] [INFO] [server.go:409] [reset-ts] [new-ts=329750308454400000] [ignore-smaller=false] [skip-upper-bound-check=false] [keyspace-group-id=0]
[2023/08/16 18:32:16.709 +08:00] [INFO] [server.go:409] [reset-ts] [new-ts=919750308454400000] [ignore-smaller=false] [skip-upper-bound-check=false] [keyspace-group-id=0]
[2023/08/16 18:32:31.409 +08:00] [INFO] [server.go:409] [reset-ts] [new-ts=919750308454400000] [ignore-smaller=true] [skip-upper-bound-check=true] [keyspace-group-id=0]
[2023/08/16 18:32:31.432 +08:00] [WARN] [tso.go:312] ["the logical time may be not enough"] [prev-logical=196608]
```

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/tso/server/server.go [hunk 1]
```diff
@@ -411,6 +404,28 @@ func (s *Server) SetExternalTS(externalTS uint64) error {
 	return nil
 }
 
+// ResetTS resets the TSO with the specified one.
+func (s *Server) ResetTS(ts uint64, ignoreSmaller, skipUpperBoundCheck bool, keyspaceGroupID uint32) error {
+	log.Info("reset-ts",
```
Comment (JmPotato) [inline pkg/mcs/tso/server/server.go:409]: Will this log be overwhelmed?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: pkg/mcs/tso/server/server.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/tso/server/server.go:409]: It is the same as the PD mode. I think it's OK?

**Review by lhy1024 — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**